### ISSUE TITLE
Scripting CodeKit?

### ISSUE THREAD

**Issue by noahfrederick** [bdkjones/CodeKit2#53]
Is there any facility for scripting CodeKit 2, either with AppleScript or through app-specific URLs? Is such a thing planned? In particular, I'm interested in:
1. Switching projects, or launching CodeKit with a particular project
2. Launching a project preview (not as important)

My particular use case is controlling CodeKit from my editor.

Feel free to close if this is not the appropriate channel for this question. Thanks!

**Comment by mdavis1982** [comment]
@noahfrederick I think this is a legitimate feature request and I'd be :+1: for adding it. I can see a few issues, however - Apple seems to be removing more and more AppleScript with every release of the OS - I'm not sure it would be worth investing the time to make CodeKit scriptable in this way.

App-specific URLs may be a way to do that, but @bdkjones would need to make that decision. I can't see this making it into the first release though, to be honest. Over to him! :grinning:

**Comment by Horttcore** [comment]
Control via terminal would also be cool, so we can use Alfred2 workflows with CodeKit.

**Comment by noahfrederick** [comment]
@Horttcore Yep. You would get that for free with either method:

```
open codekit://project/example.com
```

or

```
osascript -e "tell application CodeKit to open project...whatever"
```

@mdavis1982 (Re: AppleScript) Indeed. Such is the way of proprietary software. At the same time, though, Apple is actively adding features to the language. Very confusing.

I'm not an app developer, but I can imagine the URL method would be much simpler to implement and would cover my needs. This has my vote for a 2.x feature.

**Comment by bdkjones** [comment]
It's not at the top of my priority list for now, but I think I can do this in 2.1 or 2.2.

**Comment by bdkjones** [comment]
Implemented for 2.1.8. Thanks for sticking in there.

**Comment by noahfrederick** [comment]
> Implemented for 2.1.8. Thanks for sticking in there.

Thanks, Bryan.

For those who are interested, I finally got around to making Vim plug-in for controlling CodeKit:
https://github.com/noahfrederick/vim-codekit
